### PR TITLE
misc: bump prometheus

### PR DESCRIPTION
**Pull request by MichaHoffmann**
* bump prometheus to latest main
* fix test fallout by repairing step_invariant to deal with histograms better
* used `comparer` for all comparisons between old and new results
* dropped promql annotations in `comparer` for now

### CODE REVIEW

File: execution/step_invariant/step_invariant.go [hunk 1]
```diff
@@ -142,7 +142,7 @@ func (u *stepInvariantOperator) cacheInputVector(ctx context.Context) error {
 		}
 		defer u.next.GetPool().PutVectors(in)
 
-		if len(in) == 0 || len(in[0].Samples) == 0 {
+		if len(in) == 0 || (len(in[0].Samples) == 0 && len(in[0].Histograms) == 0) {
```
Comment (MichaHoffmann) [inline execution/step_invariant/step_invariant.go:145]: A bit of a bundled bugfix here; we can do that in its own PR i think too; wdyt?

File: engine/engine_test.go [hunk 2]
```diff
@@ -4911,6 +4911,16 @@ func roundValues(r *promql.Result) {
 	}
 }
 
+// dropAnnotations drops anntoations from the result
+// TODO: this should be done in a comparer and is only
+// because prometheus got more liberal with its usage of
+// annotations for info purposes. It should only be used
+// if we use a mock storage that doesnt send the same info
+// annotations.
+func dropAnnotations(r *promql.Result) {
```
Comment (MichaHoffmann) [inline engine/engine_test.go:4920]: We should try to mirror the annotations that PromQL adds at some point but this is a larger undertaking so ill just drop them for comparisons sake now. 
It is not optimal and we should follow-up with a PR that does this smarter.

**Review by yeya24 — APPROVED** [review]
Thanks LGTM.

File: execution/step_invariant/step_invariant.go (hunk 1 above)
Comment (yeya24) [reply execution/step_invariant/step_invariant.go:145]: I am ok to fix it this pr
